hles/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala

```diff
@@ -77,4 +86,32 @@ object HLESurveyTransformationPipelineBuilder extends PipelineBuilder[Args] {
           RawRecord(id.toLong, fields)
       }
   }
+
+  /** Read in records and group by study Id, with field name subgroups. */
+  def readEnvRecords(ctx: ScioContext, args: Args): SCollection[RawRecord] = {
+    val rawRecords: SCollection[Msg] = StorageIO
+      .readJsonLists(
+        ctx,
+        "Raw Environment Records",
+        s"${args.inputPrefix}/records/*.json"
+      )
+
+    // Group by study ID (record number) and arm (address_year_month)
+    // to get the format: (studyId, arm_id, Iterable((fieldName, Iterable(value))))
+    // (study_id, arm_id, iterable())
+    rawRecords
+      .groupBy(record => {
+        (record.read[String]("record"), record.read[String]("redcap_event_name"))
+      })
+      .map {
+        case ((id, eventName), rawRecordValues) =>
+          val fields: Map[String, Array[String]] = rawRecordValues
+            .groupBy(_.read[String]("field_name"))
+            .map {
+              case (fieldName, rawValues) =>
+                (fieldName, rawValues.map(_.read[String]("value")).toArray.sorted)
```

Comment:
this block (converting from redcap data structures to something more useful to us) seems like it's something we might want to be able to reuse. can we pull this functionality out into a `RedcapRecordDigester` (working title) object or something similar?